lib/bash/bootstrap.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+# Purpose: provide a library of reusable bash related functions
+
+function z_path_add() {
+  local -r directory="$1"
+  local -r current_path="$PATH"
+  if [ -z "$current_path" ]; then
+    export "PATH=$directory"
+  else
+    if ! echo "$PATH" | grep -- "$directory" >/dev/null 2>&1; then
+      export "PATH=$directory:$PATH"
+    fi
+  fi
+}
+
+
+if [ "${BASH_SOURCE[0]}" != "$0" ]; then
+  if [ -z "$Z_HOME" ]; then
+    export -r Z_HOME=$(dirname $(dirname $(dirname $0)))
+    export -r Z_BIN=${Z_HOME}/bin
+    export -r Z_LIB=${Z_HOME}/lib/bash
```

Comment:
Also Z_LIB might be better off defined as `${Z_HOME}/lib` instead of `${Z_HOME}/lib/bash` and then we can have a `${Z_BASH_LIB}` for the above value.